packages/react/src/inbox/useConversations.ts

```diff
@@ -0,0 +1,77 @@
+import type {
+  Conversation,
+  ConversationsDisabledError,
+  ProfileId,
+} from '@lens-protocol/domain/entities';
+import { EthereumAddress, invariant } from '@lens-protocol/shared-kernel';
+import { useEffect, useState } from 'react';
+
+import type { ReadResult } from '../helpers/reads';
+import { useGetAllConversationsController } from './adapters/useGetAllConversationsController';
+import type { InboxConfig } from './config';
+
+export type UseConversationsArgs = {
+  profileId?: ProfileId;
+  address: EthereumAddress;
+  config: InboxConfig;
+};
+
+/**
+ * @internal
+ * @experimental
+ */
+export function useConversations(
+  args: UseConversationsArgs,
+): ReadResult<Conversation[], ConversationsDisabledError> {
```

Comment:
What happened to the Conversation DTO we used to have a previous iteration?

Better not expose the entities directly... they might evolve into object with behaviour attached to them and here we just need data.

The fact they are the same today... at the beginning of building a new feature... should be considered an apparent duplication. These types changes at different speed for different reason. Better not couple them even if it saves few keystrokes today.

We can leverage TS structural typing and for now just define a new type that has same properties of the entities.